src/sentry/uptime/endpoints/validators.py

```diff
@@ -313,3 +356,167 @@ def update(self, instance: Detector, data):
             )
 
         return instance
+
+
+class UptimeMonitorDataSourceValidator(BaseDataSourceValidator[UptimeSubscription]):
+    @property
+    def data_source_type_handler(self) -> type[UptimeSubscriptionDataSourceHandler]:
+        return UptimeSubscriptionDataSourceHandler
+
+    name = serializers.CharField(
+        required=True,
+        max_length=128,
+        help_text="Name of the uptime monitor.",
+    )
+    status = serializers.ChoiceField(
+        choices=list(zip(MONITOR_STATUSES.keys(), MONITOR_STATUSES.keys())),
+        default="active",
+        help_text="Status of the uptime monitor. Disabled uptime monitors will not perform checks and do not count against the uptime monitor quota.",
+    )
+    owner = ActorField(
+        required=False,
+        allow_null=True,
+        help_text="The ID of the team or user that owns the uptime monitor. (eg. user:51 or team:6)",
+    )
+    environment = serializers.CharField(
+        max_length=64,
+        required=False,
+        allow_null=True,
+        help_text="Name of the environment to create uptime issues in.",
+    )
```

Comment:
This isn't used either right, since environment is part of the detectors `config` field right